src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3856,105 +3808,14 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
-    // WindowName is a otherwise generic WINRT_OBSERVABLE_PROPERTY, but it needs
-    // to raise a PropertyChanged for WindowNameForDisplay, instead of
-    // WindowName.
-    winrt::hstring TerminalPage::WindowName() const noexcept
-    {
-        return _WindowName;
-    }
-
-    winrt::fire_and_forget TerminalPage::WindowName(const winrt::hstring& value)
-    {
-        const auto oldIsQuakeMode = IsQuakeWindow();
-        const auto changed = _WindowName != value;
-        if (changed)
-        {
-            _WindowName = value;
-        }
-        auto weakThis{ get_weak() };
-        // On the foreground thread, raise property changed notifications, and
-        // display the success toast.
-        co_await wil::resume_foreground(Dispatcher());
-        if (auto page{ weakThis.get() })
-        {
-            if (changed)
-            {
-                page->_PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"WindowName" });
-                page->_PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"WindowNameForDisplay" });
-
-                // DON'T display the confirmation if this is the name we were
-                // given on startup!
-                if (page->_startupState == StartupState::Initialized)
-                {
-                    page->IdentifyWindow();
-
-                    // If we're entering quake mode, or leaving it
-                    if (IsQuakeWindow() != oldIsQuakeMode)
-                    {
-                        // If we're entering Quake Mode from ~Focus Mode, then this will enter Focus Mode
-                        // If we're entering Quake Mode from Focus Mode, then this will do nothing
-                        // If we're leaving Quake Mode (we're already in Focus Mode), then this will do nothing
-                        SetFocusMode(true);
-                        _IsQuakeWindowChangedHandlers(*this, nullptr);
-                    }
-                }
-            }
-        }
-    }
-
-    // WindowId is a otherwise generic WINRT_OBSERVABLE_PROPERTY, but it needs
-    // to raise a PropertyChanged for WindowIdForDisplay, instead of
-    // WindowId.
-    uint64_t TerminalPage::WindowId() const noexcept
-    {
-        return _WindowId;
-    }
-    void TerminalPage::WindowId(const uint64_t& value)
-    {
-        if (_WindowId != value)
-        {
-            _WindowId = value;
-            _PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"WindowIdForDisplay" });
-        }
-    }
-
-    void TerminalPage::SetPersistedLayoutIdx(const uint32_t idx)
-    {
-        _loadFromPersistedLayoutIdx = idx;
-    }
-
     void TerminalPage::SetNumberOfOpenWindows(const uint64_t num)
```

Comment:
now _this_ seems like something that belongs in AppLogic

Reply:
It's the most ridiculous thing. It's only ever read in one place, in `TerminalPage::_RemoveTab`

```c++
        // To close the window here, we need to close the hosting window.
        if (_tabs.Size() == 0)
        {
            // If we are supposed to save state, make sure we clear it out
            // if the user manually closed all tabs.
            // Do this only if we are the last window; the monarch will notice
            // we are missing and remove us that way otherwise.
            if (!_maintainStateOnTabClose && _settings.GlobalSettings().ShouldUsePersistedLayout() && _numOpenWindows == 1)
            {
                auto state = ApplicationState::SharedInstance();
                state.PersistedWindowLayouts(nullptr);
            }

            _LastTabClosedHandlers(*this, nullptr);
        }
```

Basically, it's used to control "if the user closed the last tab in the last window _by closing the tab_, then clear out the persisted state, so the terminal would re-open without a persisted layout"

It was discussed over here: https://github.com/microsoft/terminal/issues/9800#issuecomment-922912405